### PR TITLE
add PayPal ticket generation support

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/97806

While the "undo" is still missing the biggest functionality is there, will implement the removal of orders in another PR.

### CODE REVIEW

**Comment by lucatume** [comment]
Working on this some more

**Comment by lucatume** [comment]
I've updated the code to include the Orders removal functionality and to make it work with TC S3 Code

File: src/Commerce/Generator/PayPal/CLI.php [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+				continue;
+			}
+
+			$this->order_status = $order_status;
+
+			$progress->tick();
+
+			$faker = Faker\Factory::create();
+
+			$transaction_id = strtoupper( substr( md5( $faker->sentence ), 0, 17 ) );
```
Comment (bordoni) [inline src/Commerce/Generator/PayPal/CLI.php]: Why not use the `$faker->asciify( str_repeat( '*' ,17 ) )`? Same kind of result but "more random".

File: src/Commerce/Generator/PayPal/CLI.php [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+				'mc_handling1'           => '0.00',
+				'payment_fee'            => $this->signed_value( 0.09 ),
+				'transaction_subject'    => '',
+				'txn_id'                 => $transaction_id,
+				'residence_country'      => 'US',
+				'payment_status'         => ucwords( $order_status ),
+				'mc_fee'                 => $this->signed_value( 0.09 ),
+				'mc_gross'               => $this->signed_value( $payment_gross ),
+				'insurance_amount'       => '0.00',
+				'address_country'        => 'United States',
```
Comment (bordoni) [inline src/Commerce/Generator/PayPal/CLI.php:111]: Should we randomize the address in general?

File: src/Commerce/Generator/PayPal/CLI.php [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+			$postarr['meta_input'][ Meta_Keys::$generated_meta_key ] = 1;
+
+			return $postarr;
+		} );
+
+		// no, do not send emails to the fake attendees
+		add_filter( 'tribe_tickets_tpp_send_mail', '__return_false' );
+
+		// do not `die` after generating tickets
+		add_filter( 'tribe_exit', function () {
```
Comment (bordoni) [inline src/Commerce/Generator/PayPal/CLI.php:325]: Why a function here?

File: src/Meta_Keys.php [hunk 4]
```diff
@@ -12,4 +12,5 @@ class Tribe__Cli__Meta_Keys {
 	 * @var string
 	 */
 	public static $generated_meta_key = '_tribe_cli_generated';
+	public static $total_sales_backup_meta_key = '_tribe_cli_total_sales_backup';
```
Comment (bordoni) [inline src/Meta_Keys.php]: Docblock

**Review by bordoni — APPROVED** [review]

File: src/Commerce/Generator/PayPal/CLI.php (hunk 1 above)
Comment (lucatume) [reply src/Commerce/Generator/PayPal/CLI.php]: Because that method will use symbols too.

File: src/Commerce/Generator/PayPal/CLI.php (hunk 2 above)
Comment (lucatume) [reply src/Commerce/Generator/PayPal/CLI.php:111]: Good idea, still in the US for the time being.

File: src/Commerce/Generator/PayPal/CLI.php (hunk 3 above)
Comment (lucatume) [reply src/Commerce/Generator/PayPal/CLI.php:325]: You need to return a handler, not handle itself. Like `wp_die`.

**Comment by lucatume** [comment]
Made the changes, good for another look.

**Review by borkweb — APPROVED** [review]
:+1:

**Pull request merged**